unstructured_ingest/v2/processes/connectors/redisdb.py

```diff
@@ -0,0 +1,155 @@
+    async def create_client(self) -> AsyncGenerator["Redis", None]:
+        from redis.asyncio import Redis, from_url
+
+        access_config = self.access_config.get_secret_value()
+        options = {
+            "host": self.host,
+            "port": self.port,
+            "db": self.database,
+            "ssl": self.ssl,
+            "username": self.username,
+        }
+
+        if access_config.uri:
+            client = from_url(access_config.uri)
+        elif access_config.password:
+            options["password"] = access_config.password
+            client = Redis(**options)
+        else:
+            client = Redis(**options)
+
+        try:
+            yield client
+        finally:
+            await client.aclose()
+
+
+class RedisUploaderConfig(UploaderConfig):
+    batch_size: int = Field(default=100, description="Number of records per batch")
+
+
+@dataclass
+class RedisUploader(Uploader):
+    upload_config: RedisUploaderConfig
+    connection_config: RedisConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def is_async(self) -> bool:
+        return True
+
+    def precheck(self) -> None:
+        if not self.connection_config.access_config.uri and not self.connection_config.host:
+            raise ValueError("Please pass a hostname either directly or through uri")
+
+        async def check_connection():
+            async with self.connection_config.create_client() as async_client:
+                await async_client.ping()
+
+        try:
+            asyncio.run(check_connection())
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    @requires_dependencies(["redis"], extras="redis")
+    async def run_async(self, path: Path, file_data: FileData, **kwargs: Any) -> None:
+        from redis import exceptions as redis_exceptions
+
+        with path.open("r") as file:
+            elements_dict = json.load(file)
+
+        if elements_dict:
+            logger.info(
+                f"writing {len(elements_dict)} objects to destination asynchronously, "
+                f"db, {self.connection_config.database}, "
+                f"at {self.connection_config.host}",
+            )
+            async with self.connection_config.create_client() as async_client:
+                async with async_client.pipeline(transaction=True) as pipe:
+                    first_element = elements_dict[0]
+                    element_id = first_element["element_id"]
+                    redis_stack = True
+                    try:
+                        # Redis with stack extension supports JSON type
+                        await pipe.json().set(element_id, "$", first_element).execute()
+                    except redis_exceptions.ResponseError as e:
+                        message = str(e)
+                        if "unknown command `JSON.SET`" in message:
```

Comment:
there's no way to pull this from the redis cluster before hand? By querying metadata perhaps? Just wondering, if not, this is fine.

Reply:
I checked all the commands regarding the cluster, but don't see info whether it has stack extension or supports JSON.
https://redis.io/docs/latest/commands/cluster-info/